test/transportSpec.js

```diff
@@ -0,0 +1,518 @@
+'use strict'
+
+const co = require('co')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+chai.use(chaiAsPromised)
+const assert = chai.assert
+const expect = chai.expect
+const moment = require('moment')
+
+const ILP = require('..')
+const Transport = require('../src/lib/transport')
+const Packet = require('../src/utils/packet')
+const MockPlugin = require('./mocks/mockPlugin')
+const { wait } = require('../src/utils')
+const base64url = require('../src/utils/base64url')
+const { parsePacketAndDetails } = require('../src/utils/details')
+const Details = require('../src/utils/details')
+
+describe('Transport', function () {
+  describe('PSK', function () {
+    beforeEach(function () {
+      this.params = {
+        destinationAccount: 'test.example.alice',
+        secretSeed: Buffer.from('shh_its_a_secret', 'base64')
+      }
+    })
+
+    it('should generate psk params', function () {
+      const params = ILP.PSK.generateParams(this.params)
+
+      assert.match(params.destinationAccount, /^test\.example\.alice/)
```

Comment:
This should also check that the way it generates the shared secret is consistent with the spec. It should test against a hard-coded value, possibly using reRequire to mock crypto.randomBytes so that it always returns the same thing in testing.